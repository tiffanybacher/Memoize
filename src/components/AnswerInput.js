import React, { Component } from 'react';

class AnswerInput extends Component {
  constructor(props) {
    super(props);
  }

  handleChange = (e) => {
    
  }

  handleClick = () => {
    this.props.findAnswer();
    this.props.getUserAnswer(this.userInput.innerText);
  }

  render() {
    return (
      <article className="AnswerInput">
        <h3>Answer:</h3>
        <div className="flex-container">
          <div 
            role="textbox"
            className="input"
            contentEditable="true"
            ref={el => (this.userInput = el)}
          >
          </div>
        </div>
          <div
            role="button"
            className="submit-btn"
            onClick={this.handleClick}>
            Submit 
          </div>
      </article>
    );
  }
}

export default AnswerInput;