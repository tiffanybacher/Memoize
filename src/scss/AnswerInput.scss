.AnswerInput {
  margin: 0 auto;

  h3 {
    margin: 0 0 10px;
  }

  .flex-container {
    display: flex;
    justify-content: center;
  }

  .input {
    background-color: white;
    border-radius: $rounded-border;
    border: 4px solid #999;
    color: #333;
    font-size: 1.2em;
    line-height: 1.5;
    overflow: auto;
    padding: 10px 15px;
    width: 100%;

    &:hover {
      border: $purple-border;
      cursor: text;
    }
    
    &:focus {
      border: $purple-border;
    }
  }
}
