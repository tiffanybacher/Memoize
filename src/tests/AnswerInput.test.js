import React from 'react';
import ReactDom from 'react-dom';
import { shallow } from 'enzyme';
import AnswerInput from '../components/AnswerInput';

describe('AnswerInput', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <AnswerInput />
    );
  });

  it('should match snapshot', () +> {
    expect(wrapper).toMatchSnapshot();
  });
});