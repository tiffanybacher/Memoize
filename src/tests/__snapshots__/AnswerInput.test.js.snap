// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerInput should match snapshot 1`] = `
<article
  className="AnswerInput"
>
  <h3>
    What's your guess?
  </h3>
  <div
    className="flex-container"
  >
    <div
      className="input"
      contentEditable="true"
      onInput={[Function]}
      role="textbox"
    />
  </div>
  <div
    className="submit-btn"
    onClick={[Function]}
    role="button"
  >
    Submit
  </div>
</article>
`;
